pdns/ixfrdist-stats.hh

```diff
@@ -0,0 +1,102 @@
+/*
+ * This file is part of PowerDNS or dnsdist.
+ * Copyright -- PowerDNS.COM B.V. and its contributors
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of version 2 of the GNU General Public License as
+ * published by the Free Software Foundation.
+ *
+ * In addition, for the avoidance of any doubt, permission is granted to
+ * link this program with OpenSSL and to (re)distribute the binaries
+ * produced as the result of such linking.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+#pragma once
+#include <atomic>
+#include <map>
+#include <string>
+
+#include "dnsname.hh"
+#include "pdnsexception.hh"
+
+class ixfrdistStats {
+  public:
+    ixfrdistStats() {
+      progStats.startTime = time(nullptr);
+    }
+
+    std::string getStats();
+
+    void setSOASerial(const DNSName& d, const uint32_t serial) {
+      wasRegistered(d);
+      domainStats[d].currentSOA = serial;
+      domainStats[d].haveZone = true;
```

Comment:
It's a bit sad to do the lookup in the map twice every time you update the statistics, how about having `wasRegistered()` return the iterator?

Reply:
fixed!